lib/fog/ovirt/models/compute/template.rb

```diff
@@ -42,11 +42,15 @@ def ready?
 
         def destroy(options = {})
           service.client.destroy_template(id)
+        rescue ::OVIRT::OvirtException => e
+          raise ::Fog::Ovirt::OvertError, e.message
         end
 
         def save
-          raise Fog::Errors::Error.new('Resaving an existing object may create a duplicate') if persisted?
+          raise ::Fog::Ovirt::Errors::OvirtError, 'Providing an existing object may create a duplicate object' if persisted?
           service.client.create_template(attributes)
+        rescue ::OVIRT::OvirtException => e
```

Comment:
Can we wrap the `client` with a generic exception handler? It would be nicer, and future client calls would be covered automatically.

I mean something like:
``` ruby
class ClientWrapper
  def initialize(client)
     @actual_client = client
  end

  def method_missing(method, *params)
    @actual_clent.send(method, *params)
    rescue ::OVIRT::OvirtException => e
      raise ::Fog::Ovirt::OvirtError, e
    end
  end
end
```

Reply:
At the moment it does not work like that.
there is a definition of `@client` that is an ovirt sdk client, and a method that does a read only property for it named `client`.

Each place I went to, opens that specific class, and add a client.
If I understand what you are suggesting, it means to rewrite the whole concept of how it works in order to have the same way of doing stuff. But that's task number two, to my recollection.

Reply:
My bad, this comment belongs to request class, not the model. I am talking about wrapping the [client variable](https://github.com/ik5/fog-ovirt/blob/8c311047a9e597fa29a1614196891291454b0ab6/lib/fog/ovirt/compute.rb#L143) in the request, so calls to client won't emit `::OVIRT::OvirtException` at all.

Reply:
done